app/models/course.rb

```diff
@@ -20,6 +20,54 @@ class Course < ActiveRecord::Base
   delegate :staff, to: :course_users
   delegate :has_user?, to: :course_users
 
+  # Generates the level numbers for the Course::Levels,
+  # since only the thresholds are stored in database.
+  # The entries should have been sorted during the
+  # database query.
+  #
+  # @param levels_collection A CollectionProxy containing Course::Levels
+  # @return The CollectionProxy with numbered Course::Levels
+  def number_levels(levels_collection)
+    levels_collection.each_with_index do |level, index|
```

Comment:
`levels_collection.map...` because you want the result, not calling the block for effect.

This stuff ought to be in a concern too, methinks.